src/Toml/Bi/Combinators.hs

```diff
@@ -513,3 +513,33 @@ map keyCodec valCodec key = Codec input output
                     insertTableArrays key $ t :| (ts ++ tomls)
 
         dict <$ modify updateAction
+
+tableMapCodec
+    :: Ord key
+    => TomlBiMap Key key  -- ^ Bidirectional converter between TOML and Map keys
+    -> TomlBiMap val AnyValue  -- ^ Bidirectional converter between TOML and Map values
+    -> Key  -- ^ Table name key
+    -> TomlCodec (Map key val)
+tableMapCodec keyBiMap valBiMap key = Codec input output
+  where
+    -- input :: Env (Map key val)
+    input = do
+        mTable <- asks $ Prefix.lookup key . tomlTables
+        case mTable of
+            Nothing   -> pure Map.empty
+            Just toml -> fmap Map.fromList $ forM (HashMap.toList (tomlPairs toml)) $ \(tomlKey, anyVal) ->
+                case backward valBiMap anyVal of
+                    Right v  -> case forward keyBiMap tomlKey of
+                        Right k  -> pure (k, v)
+                        Left err -> throwError $ BiMapError err
+                    Left err -> throwError $ BiMapError err
+
+    -- output :: Map key val -> St (Map key val)
+    output dict = do
+        newToml <- foldM update mempty (Map.toList dict)
+        dict <$ modify (insertTable key newToml)
+
+    update toml (k, v) = do
+        tomlKey <- MaybeT $ pure $ either (const Nothing) Just $ backward keyBiMap k
+        anyVal <- MaybeT $ pure $ either (const Nothing) Just $ forward valBiMap v
```

Comment:
Yeah, this looks a bit verbose... It could be written shorter, if a library provided a function like:

```haskell
eitherToMaybeT :: Applicative m => Either e a -> MaybeT m a
```

But [Hoogle](https://hoogle.haskell.org/?hoogle=Either+e+a+-%3E+MaybeT+m+a&scope=set%3Astackage) doesn't show anything 🤔

Reply:
I basically lifted that form directly from the `match` function. So if there is a solution it would be useful there as well.